src/etc/systemd/system/kea-dhcp4-server.service.d/override.conf

```diff
@@ -5,3 +5,5 @@ After=vyos-router.service
 [Service]
 ExecStart=
 ExecStart=/usr/sbin/kea-dhcp4 -c /run/kea/kea-dhcp4.conf
+ExecStartPost=!/usr/bin/python3 /usr/libexec/vyos/op_mode/dhcp.py update_dhcp_server_lease_to_hostd_state --family inet
```

Comment:
Thanks for the review @sarthurdev! I need some feedback on the convention of invoking this one.
So far I haven't seen any precedence of invoking `op_mode/*.py` directly in the codebase. Is this okay, or do you want me to revisit this?

Reply:
it's not the ideal solution, as I don't see a use case where it'd be needed in op-mode, nor is it defined in XML.

Probably best done in a script in `src/system` like the Kea hook scripts.

Reply:
Yes, that's why I asked :)

As you can see in the implementation of `update_dhcp_server_lease_to_hostd_state`, the method crucially depends on `_get_dhcp_pools` and `_get_raw_server_leases` which are implemented in `op_mode/dhcp.py` which is why I added the method in `update_dhcp_server_lease_to_hostd_state`. Could they be refactored to kea lib (`python/vyos/kea.py`) perhaps so that they can be reused?

Or what if we actually add an op-mode command - something like `update hostd state` and define it in XML as well?

NB: I realize that we are getting into design discussion here. We can discuss the options on Slack if necessary.

Reply:
Ideally there shouldn't be a scenario where the user needs to manually re-sync hostd state.

I agree that the utility functions could move to the `vyos.kea` module and be imported in op-mode and a new src/system script. Though for the module function, it should probably be just the logic that needs to be re-used (no sorting, filtering etc. Just returning the lease data as-is).

Reply:
@sarthurdev, raised #4307 to refactor utility functions to `vyos.kea`. Would appreciate feedback on that.

Reply:
Now that #4307 has been applied, I have made the necessary changes to better comply with the recommended convention.